### PR TITLE
EJB Runtime Statistics

### PR DESCRIPTION
**Pull request by spriadka**
This PR brings tests for viewing runtime EJB statistics.

### CODE REVIEW

**Comment by spriadka** [comment]
There are still some failures in tests which need to be resolved before merging, it is ``MessageDrivenEJBTest.verifyExecutionTime``, ``MessageDrivenEJBTest.verifyInvocations``, ``MessageDrivenEJBTest.verifyPeakConcurrentInvocations``, ``SingletonEJBTest.verifyExecutionTime``, ``StatefulEJBTest.verifyPeakConcurrentInvocations``. As the amount of files in this PR is quite big, any tips/comments/advices are welcome

File: tests/basic/src/test/java/org/jboss/hal/testsuite/test/runtime/ejb/ejb/MessageDrivenEJBExecutionTime.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    @ActivationConfigProperty(propertyName = "acknowledgeMode", propertyValue = "Auto-acknowledge")})
+public class MessageDrivenEJBExecutionTime implements MessageListener {
+
+    private final Logger logger = Logger.getLogger("MessageDrivenEJBExecutionTime");
+
+    @Override
+    public void onMessage(Message message) {
+        TextMessage msg;
+        try {
+            Thread.sleep(5000);
```
Comment (zebrik) [inline tests/basic/src/test/java/org/jboss/hal/testsuite/test/runtime/ejb/ejb/MessageDrivenEJBExecutionTime.java]: It's quite long. How many time is this sleep called? Such long sleep appears 9 times in the code of this pull request.

File: tests/basic/src/test/java/org/jboss/hal/testsuite/test/runtime/ejb/MessageDrivenEJBTest.java [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+            .verifyExists()
+            .verifyAttribute("run-as-role", EJBFixtures.ROLE_1);
+    }
+
+    private static void invoke(int numberOfInvocations, EJBDeployment ejbDeployment) {
+        String deploymentName = ejbDeployment.getDeploymentName().substring(0,
+            ejbDeployment.getDeploymentName().indexOf(".war"));
+        HttpGet httpGet;
+        try {
+            URL url = new URL("http://0.0.0.0:8080/" + deploymentName + "/MessageDriven");
```
Comment (zebrik) [inline tests/basic/src/test/java/org/jboss/hal/testsuite/test/runtime/ejb/MessageDrivenEJBTest.java]: Hardcoded port. Maybe we can start thinking about adding the port offset to the testsuite configuration. Just note, not needed to be fixed immediately.

File: tests/basic/src/test/java/org/jboss/hal/testsuite/test/runtime/ejb/MessageDrivenEJBTest.java [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+            page.getMessageDrivenBeanEJBForm().value("run-as-role"));
+        new ResourceVerifier(
+            EJBFixtures.messageDrivenEJBAddress(MESSAGE_DRIVEN_EJB_WITH_ROLES_DEPLOYMENT_NAME,
+                MessageDrivenEJBWithRoles.class),
+            client)
+            .verifyExists()
+            .verifyAttribute("run-as-role", EJBFixtures.ROLE_1);
+    }
+
+    private static void invoke(int numberOfInvocations, EJBDeployment ejbDeployment) {
```
Comment (zebrik) [inline tests/basic/src/test/java/org/jboss/hal/testsuite/test/runtime/ejb/MessageDrivenEJBTest.java]: Seems similar method is written more times. What about moving this method to some library class? WDYT?

File: tests/basic/src/test/java/org/jboss/hal/testsuite/test/runtime/ejb/ejb/MessageDrivenEJB.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+                msg = (TextMessage) message;
+                logger.info("Received Message from MessageDrivenQueue: " + msg.getText());
+            } else {
+                logger.warn("Message of wrong type: " + message.getClass().getName());
+            }
+        } catch (JMSException | InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+}
```
Comment (zebrik) [inline tests/basic/src/test/java/org/jboss/hal/testsuite/test/runtime/ejb/ejb/MessageDrivenEJB.java]: You can add a new line at the end of the file. Can be applied to more files in this pull request.

**Comment by spriadka** [comment]
Thank you for the feedback @zebrik, I have moved the ``client.close()`` method higher in the abstraction, made the ``invoke(...)`` abstract and reduced the sleep times. I have added the hardcoded port thing for further work.

**Pull request merged**